src/wallet/spend.cpp

```diff
@@ -450,6 +466,18 @@ CoinsResult AvailableCoins(const CWallet& wallet,
             type = Solver(script, script_solutions);
             is_from_p2sh = true;
         }
+        // Very unlikely we'd be spending a witness unknown output, but if we are trying to pay a
+        // silent payments v0 address, this can't be included
+        if (silent_payment && type == TxoutType::WITNESS_UNKNOWN) continue;
```

Comment:
https://github.com/bitcoin/bitcoin/pull/28201/commits/e0404435ff4d64b32b58774b6c909c66f1692a9f: `TxoutType::PUBKEY` which represents Pay2Pubkey outputs, is also not eligible to pay to silent payments outputs.

Reply:
Makes sense